contrib/brl/bpro/core/vpgl_pro/processes/vpgl_get_bounding_box_process.cxx

```diff
@@ -88,10 +88,10 @@ bool vpgl_get_bounding_box_process(bprb_func_process& pro)
 
     //intersect each ray with z plane
     vgl_point_3d<double> ulp, urp, blp, brp;
-    bool good =    vgl_intersection(ul, zp, ulp);
-    good = good && vgl_intersection(ur, zp, urp);
-    good = good && vgl_intersection(bl, zp, blp);
-    good = good && vgl_intersection(br, zp, brp);
+    vgl_intersection(ul, zp, ulp);
+    vgl_intersection(ur, zp, urp);
+    vgl_intersection(bl, zp, blp);
+    vgl_intersection(br, zp, brp);
 
```

Comment:
David,

That is a good point,   but there is a limit to the  amount of effort we can put into this.  We have expertise in removing problematic code that has been there for a long time, but not in the complete code base.

To be clear, our goal is to remove these “noisy” warnings so that we can more effectively treat the “real warnings”  There are hundreds of other warnings are are more indicative of bugs or other logical errors.

I will ask @ajjl to revert this change and improve the test, but let’s try to find a balance between “optimal code changes” and “evolutionary code changes”.

========

Thank you for your quick response.
-- 


From:  Daniel Crispell <notifications@github.com>
Reply-To:  vxl/vxl <reply@reply.github.com>
Date:  Thursday, February 11, 2016 at 8:22 AM
To:  vxl/vxl <vxl@noreply.github.com>
Cc:  Hans Johnson <hans.j.johnson@gmail.com>
Subject:  Re: [vxl] COMP: Fixes "Value stored to '' never read" warnings (#172)


In contrib/brl/bpro/core/vpgl_pro/processes/vpgl_get_bounding_box_process.cxx:
> @@ -88,10 +88,10 @@ bool vpgl_get_bounding_box_process(bprb_func_process& pro)
>  
>      //intersect each ray with z plane
>      vgl_point_3d<double> ulp, urp, blp, brp;
> -    bool good =    vgl_intersection(ul, zp, ulp);
> -    good = good && vgl_intersection(ur, zp, urp);
> -    good = good && vgl_intersection(bl, zp, blp);
> -    good = good && vgl_intersection(br, zp, brp);
> +    vgl_intersection(ul, zp, ulp);
> +    vgl_intersection(ur, zp, urp);
> +    vgl_intersection(bl, zp, blp);
> +    vgl_intersection(br, zp, brp);
>  
Wouldn't it be better to keep the return values from vgl_intersection, but add code to actually check it and report an error if it fails?

—
Reply to this email directly or view it on GitHub.